ui-ngx/src/app/shared/models/widget-settings.models.ts

```diff
@@ -248,27 +325,99 @@ type ValueColorFunction = (value: any) => string;
 
 export abstract class ColorProcessor {
 
-  static fromSettings(color: ColorSettings): ColorProcessor {
+  static fromSettings(color: ColorSettings, ctx?: WidgetContext, cd?: ChangeDetectorRef, maxValue = 100): ColorProcessor {
```

Comment:
It might be better to add the argument maxValue to the ColorGradientSettings interface?

Reply:
Why can we only set the maximum value?